File: fsharp/cars-assemble/CarsAssemble.fs
module CarsAssemble

open System

let hourlyProduction = 221

let successRate (speed: int): float =
    match speed with
    | s when s < 1 -> 0.0
    | s when s <= 4 -> 1.0
    | s when s <= 8 -> 0.9
    | s when s = 9 -> 0.8
    | _ -> 0.77

let productionRatePerHour (speed: int): float =
    (float speed) * (float hourlyProduction) * (successRate speed)

let workingItemsPerMinute (speed: int): int =
    (int (Math.Floor ((productionRatePerHour speed) / 60.0)))

File: java/cars-assemble/src/main/java/CarsAssemble.java
public class CarsAssemble {
    public double productionRatePerHour(int speed) {
        if (speed < 1) {
            return 0;
        }

        var efficiency = 0.77;
        if (speed < 5) {
            efficiency = 1.0;
        } else if (speed < 9) {
            efficiency = 0.9;
        } else if (speed < 10) {
            efficiency = 0.8;
        }

        return speed * 221 * efficiency;
    }

    public int workingItemsPerMinute(int speed) {
        return (int)productionRatePerHour(speed) / 60;
    }
}
